### PR TITLE
Dune port + opam metadata 2 + mustache-unix package

### PR DESCRIPTION
**Pull request by avsm**
Am using mustache templates for the new ocamllabs website, and there's this cool new build system in town...

### CODE REVIEW

File: examples/dune [hunk 1]
```diff
@@ -0,0 +1,16 @@
+(executable
+ (name mustache_example)
+ (modules mustache_example)
+ (libraries mustache ezjsonm))
+
+(executable
+ (name mustache_cli)
+ (public_name mustache-cli)
```
Comment (rgrinberg) [inline examples/dune]: If this exe is public, I think `examples` is not a good source directory for it anymore.

**Comment by rgrinberg** [comment]
I gave you merge rights @avsm. Feel free to merge once you think this is ready.

By the way, I think `mustache-unix` made more sense if we had a `mustache-unix` library as well. If it's just a binary, including it in `mustache` will probably do no harm.

File: examples/dune (hunk 1 above)
Comment (anton-trunov) [reply examples/dune]: @rgrinberg Is this what needs to be done here? https://github.com/anton-trunov/ocaml-mustache/commit/82957c456d87c81bf2a108ce4d66b150943c7fd9

Comment (anton-trunov) [reply examples/dune]: @rgrinberg Thanks for the invitation! Let me push those changes here.

**Comment by anton-trunov** [comment]
It seems that the tests only fail for OCaml 4.02. On my machine the tests pass:

**Comment by rgrinberg** [comment]
You are free to drop support for 4.02 if you wish.

**Comment by anton-trunov** [comment]
> You are free to drop support for 4.02 if you wish.

Done. Now CI is green :)

Could you merge this if you are OK with the changes?

**Comment by rgrinberg** [comment]
Sure, feel free to make a release if you want.

**Comment by anton-trunov** [comment]
Great, 3.0.3 then?

**Comment by anton-trunov** [comment]
Also, should `mustache-unix` be published as a separate `opam` package?

**Comment by rgrinberg** [comment]
My opinion about mustache-unix be a separate package is not as strong anymore. I think that having a `mustache.unix` sub library would work just as well for mirage. The command line utility being present does not hurt any portability either.

In summary: I don't have a strong opinion on this.

> Great, 3.0.3 then? Or, perhaps better 3.1.0?

If new features were introduced, then I suggest 3.1.0

**Comment by anton-trunov** [comment]
> If new features were introduced, then I suggest 3.1.0

**Comment by anton-trunov** [comment]
> My opinion about mustache-unix be a separate package is not as strong anymore. I think that having a mustache.unix sub library would work just as well for mirage. 

I'm not sure I understand how to map this notion of sublibrary on opam files. AFAIU, we need to somehow merge the two `opam` files into one, right? This seems to require using optional dependencies, but I'm not sure. Could you give any pointers on how this might be done?

**Comment by rgrinberg** [comment]
I don't think it would require optional dependencies. Basically, you would just need to move everything to one package but keep the fine grained deps on the libs themselves. That seems to be portable enough. It's not necessary though, so just something to consider.

**Pull request merged**